dtfs-central-api/api-tests/v1/portal-facility-amendments/amendment-delete.api-test.ts

```diff
@@ -0,0 +1,179 @@
+
+      expect(status).toEqual(HttpStatusCode.BadRequest);
+
+      expect(body).toEqual({
+        message: "Expected path parameter 'amendmentId' to be a valid mongo id",
+        code: API_ERROR_CODE.INVALID_MONGO_ID_PATH_PARAMETER,
+      });
+    });
+
+    it(`should return ${HttpStatusCode.NotFound} when the facility does not exist`, async () => {
+      const aValidButNonExistentFacilityId = new ObjectId().toString();
+
+      const { body, status } = (await testApi
+        .remove({ auditDetails: generatePortalAuditDetails(portalUserId) })
+        .to(generateUrl(aValidButNonExistentFacilityId, amendmentId))) as ErrorResponse;
+
+      expect(status).toEqual(HttpStatusCode.NotFound);
+      expect(body).toEqual({
+        status: HttpStatusCode.NotFound,
+        message: `Amendment not found: ${amendmentId} on facility: ${aValidButNonExistentFacilityId}`,
+      });
+    });
+
+    it(`should return ${HttpStatusCode.NotFound} when the amendment does not exist`, async () => {
+      const aValidButNonExistentAmendmentId = new ObjectId().toString();
+
+      const { body, status } = (await testApi
+        .remove({ auditDetails: generatePortalAuditDetails(portalUserId) })
+        .to(generateUrl(facilityId, aValidButNonExistentAmendmentId))) as ErrorResponse;
+
+      expect(status).toEqual(HttpStatusCode.NotFound);
+      expect(body).toEqual({
+        status: HttpStatusCode.NotFound,
+        message: `Amendment not found: ${aValidButNonExistentAmendmentId} on facility: ${facilityId}`,
+      });
+    });
+
+    it(`should return ${HttpStatusCode.NoContent} when the payload is valid & the amendment exists`, async () => {
+      const { status } = (await testApi
+        .remove({ auditDetails: generatePortalAuditDetails(portalUserId) })
+        .to(generateUrl(facilityId, amendmentId))) as ErrorResponse;
+
+      expect(status).toEqual(HttpStatusCode.NoContent);
+    });
+
+    it('should overwrite any existing amendment', async () => {
+      // Arrange
+      const existingAmendment = await createPortalFacilityAmendment({ facilityId, dealId, userId: portalUserId });
+      const existingAmendmentId = existingAmendment.amendmentId.toString();
+
+      // Act
+      const { status } = (await testApi
+        .remove({ auditDetails: generatePortalAuditDetails(portalUserId) })
+        .to(generateUrl(facilityId, existingAmendmentId))) as ErrorResponse;
+
+      expect(status).toEqual(HttpStatusCode.NoContent);
+
+      // Assert
+      const getExistingAmendmentResponse = await testApi.get(`/v1/portal/facilities/${facilityId}/amendments/${existingAmendmentId}`);
+
+      expect(getExistingAmendmentResponse.status).toEqual(HttpStatusCode.NotFound);
+      expect(getExistingAmendmentResponse.body).toEqual({
+        status: HttpStatusCode.NotFound,
+        message: `Amendment not found: ${existingAmendmentId} on facility: ${facilityId}`,
+      });
+    });
```

Comment:
I think this can be simplified
```suggestion
    it('should delete the amendment', async () => {
      // Act
      const { status } = (await testApi
        .remove({ auditDetails: generatePortalAuditDetails(portalUserId) })
        .to(generateUrl(facilityId, amendmentId))) as ErrorResponse;

      expect(status).toEqual(HttpStatusCode.NoContent);

      // Assert
      const getAmendmentResponse = await testApi.get(`/v1/portal/facilities/${facilityId}/amendments/${amendmentId}`);

      expect(getAmendmentResponse.status).toEqual(HttpStatusCode.NotFound);
      expect(getAmendmentResponse.body).toEqual({
        status: HttpStatusCode.NotFound,
        message: `Amendment not found: ${amendmentId} on facility: ${facilityId}`,
      });
    });
```